src/runtime_src/core/common/info_platform.cpp

```diff
@@ -155,6 +155,20 @@ add_performance_info(const xrt_core::device* device, ptree_type& pt)
   pt.add("performance_mode", xq::performance_mode::parse_status(mode));
 }
 
+void
+add_host_mem_info(const xrt_core::device* device, ptree_type& pt)
+{
+  std::string data_retention_string = "not supported";
+  try {
+    auto value = xrt_core::device_query<xrt_core::query::enabled_host_mem>(device);
+    data_retention_string = (value > 0 ? "enabled" : "disabled");
+  }
+  catch (xrt_core::query::exception&) {
+    // Device does not support host memory features
+  }
```

Comment:
Shouldn't this be `no_such_key` ?

Reply:
I think it should be. In the future I will start converting these over to use that instead.